### PR TITLE
steps: add build_type RelWithDebInfoAndAssert

### PR DESCRIPTION
**Pull request by nico-canta**
Here I included the new `build_type` and set the default value `default="RelWithDebInfoAndAssert"`.
There are also some old (and unrelated) changes.  Let me know if they could be useful or should be removed from this PR.

### CODE REVIEW

File: var/spack/repos/builtin/packages/neuron/package.py [hunk 1]
```diff
@@ -164,6 +164,7 @@ def cmake_enable_option(spec_requiremement):
         if "+coreneuron" in self.spec:
             args.append('-DCORENEURON_DIR=' + self.spec["coreneuron"].prefix)
 
+        args.append('-DCMAKE_PREFIX_PATH=/usr/X11')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/neuron/package.py]:
```suggestion
```

**Review by matz-e — CHANGES_REQUESTED** [review]

**Comment by matz-e** [comment]
I'd prefer if the PR title starts with the relevant package name, so here `steps: `, too :)

**Comment by nico-canta** [comment]
`var/spack/repos/builtin/packages/steps/package.py:32: [E501] line too long (99 > 79 characters)`

I don't understand this error. I can see other lines that are longer than 79 characters. Is it something else?

**Comment by nico-canta** [comment]
@tristan0x I set the default `build_type` to "Release" also here.

**Comment by matz-e** [comment]
> `var/spack/repos/builtin/packages/steps/package.py:32: [E501] line too long (99 > 79 characters)`
> 
> I don't understand this error. I can see other lines that are longer than 79 characters. Is it something else?

I think sha lines are allowed to exceed that. But this particular line can be broken up properly.

File: var/spack/repos/builtin/packages/gmsh/package.py [hunk 2]
```diff
@@ -19,6 +19,7 @@ class Gmsh(CMakePackage):
     homepage = 'http://gmsh.info'
     url = 'http://gmsh.info/src/gmsh-4.4.1-source.tgz'
 
+    version('4.8.4', sha256='760dbdc072eaa3c82d066c5ba3b06eacdd3304eb2a97373fe4ada9509f0b6ace')
```
Comment (tristan0x) [inline var/spack/repos/builtin/packages/gmsh/package.py]: Should not be part of this change.
```suggestion
```

**Review by tristan0x — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/steps/package.py [hunk 3]
```diff
@@ -31,6 +31,9 @@ class Steps(CMakePackage):
     variant("bundle", default=False, description="Use bundled libraries")
     variant("stochtests", default=True, description="Add stochastic tests to ctests")
     variant("timemory", default=False, description="Add timemory API to instrument time/memory")
+    variant("build_type", default="Release", description="CMake build type",
+            values=("Debug", "Release", "RelWithDebInfo", "MinSizeRel",
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/steps/package.py]:
```suggestion
            values=("Debug", "Release", "RelWithDebInfo", "MinSizeRel",
```

**Review by matz-e — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/steps/package.py [hunk 4]
```diff
@@ -31,6 +31,9 @@ class Steps(CMakePackage):
     variant("bundle", default=False, description="Use bundled libraries")
     variant("stochtests", default=True, description="Add stochastic tests to ctests")
     variant("timemory", default=False, description="Add timemory API to instrument time/memory")
+    variant("build_type", default="Release", description="CMake build type",
```
Comment (tristan0x) [inline var/spack/repos/builtin/packages/steps/package.py]: Keep previous spack default
```suggestion
    variant("build_type", default="RelWithDebInfo", description="CMake build type",
```

**Review by tristan0x — CHANGES_REQUESTED** [review]

**Comment by matz-e** [comment]
Could you rebase on top of the latest `develop`, please?

**Review by matz-e — APPROVED** [review]

**Review by tristan0x — APPROVED** [review]
LGTM

**Pull request merged**